cpp/src/arrow/engine/substrait/extension_set.cc

```diff
@@ -670,10 +671,10 @@ struct ExtensionIdRegistryImpl : ExtensionIdRegistry {
 template <typename Enum>
 class EnumParser {
  public:
-  explicit EnumParser(const std::vector<std::string>& options) {
+  explicit EnumParser(const std::vector<std::string>& options, int offset) {
```

Comment:
This is a good cleanup.  However, the need for `kUnspecified` came from a time when Substrait handled enums differently.  Another potential change, that might be simpler, would be to remove `kUnspecified` from the other enums.